test/api/api_test.go

```diff
@@ -236,41 +233,59 @@ func TestUserTierCache(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	// Sleep for a short time in order to make sure that the background
-	// goroutine that updates user's quotas has had time to run.
-	time.Sleep(2 * time.Second)
-	// We expect to get TierAnonymous.
-	_, b, err = at.Get("/user/limits", nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = json.Unmarshal(b, &ul)
+	// We need to try this several times because we'll only get the right result
+	// after the background goroutine that updates user's quotas has had time to
+	// run.
+	err = build.Retry(10, 200*time.Millisecond, func() error {
+		// We expect to get tier with name and id matching TierPremium20 but with
+		// speeds matching TierAnonymous.
+		_, b, err = at.Get("/user/limits", nil)
+		if err != nil {
+			return errors.AddContext(err, "failed to call /user/limits")
+		}
+		err = json.Unmarshal(b, &ul)
```

Comment:
Same as below. These lines just keep adding up. Especially if the unmarshaling also doesn't happen within `Get` which you could consider. I think PJ recently refactored the blocker in that regard.